samples/amber.cc

```diff
@@ -116,9 +119,32 @@ bool ParseArgs(const std::vector<std::string>& args, Options* opts) {
         return false;
       }
       opts->spv_env = args[i];
-
     } else if (arg == "-h" || arg == "--help") {
       opts->show_help = true;
+    } else if (arg == "-v") {
+      ++i;
+      if (i >= args.size()) {
+        std::cerr << "Missing value for -v argument." << std::endl;
+        return false;
+      }
+      const std::string& ver = args[i];
+
+      size_t dot_pos = 0;
```

Comment:
gosh, I would reach for sscanf first.

  int major=0, minor=0;
  const auto num_conversions = sscanf(args[i], "%d.%d%c", &major, minor);
  if ( num_conversions == 3 ) { ... error: bad suffix }
  if ( num_conversions < 2 ) { error: invalid parse.}
  else ... now you have major and minor..

But hey, I'm old school.

Reply:
Heh, but this is fancy and cool.